dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -348,40 +345,33 @@ private static ArtifactResults generateArtifactReport(
         new FileOutputStream(outputFile), StandardCharsets.UTF_8)) {
 
       // includes all versions
-      DependencyGraph completeDependencies = artifactInfo.getCompleteDependencies();
-      List<Update> convergenceIssues = completeDependencies.findUpdates();
+      DependencyGraph graph = artifactInfo.getCompleteDependencies();
+      List<Update> convergenceIssues = graph.findUpdates();
 
       // picks versions according to Maven rules
       DependencyGraph transitiveDependencies = artifactInfo.getTransitiveDependencies();
 
       Map<Artifact, Artifact> upperBoundFailures =
-          findUpperBoundsFailures(completeDependencies.getHighestVersionMap(),
-              transitiveDependencies);
+          findUpperBoundsFailures(graph.getHighestVersionMap(), transitiveDependencies);
 
       Map<Artifact, Artifact> globalUpperBoundFailures = findUpperBoundsFailures(
           collectLatestVersions(globalDependencies), transitiveDependencies);
 
-      List<DependencyPath> dependencyPaths = completeDependencies.list();
-
       long totalLinkageErrorCount =
           symbolProblemTable.values().stream()
               .flatMap(problemToClasses -> problemToClasses.keySet().stream())
               .distinct()
               .count();
 
-      ListMultimap<DependencyPath, DependencyPath> dependencyTree =
-          DependencyTreeFormatter.buildDependencyPathTree(dependencyPaths);
       Template report = configuration.getTemplate("/templates/component.ftl");
 
-      DependencyPath rootNode = Iterables.getFirst(dependencyTree.values(), null);
       Map<String, Object> templateData = new HashMap<>();
       templateData.put("artifact", artifact);
       templateData.put("updates", convergenceIssues);
       templateData.put("upperBoundFailures", upperBoundFailures);
       templateData.put("globalUpperBoundFailures", globalUpperBoundFailures);
       templateData.put("lastUpdated", LocalDateTime.now());
-      templateData.put("dependencyTree", dependencyTree);
-      templateData.put("dependencyRootNode", rootNode);
+      templateData.put("dependencyGraph", graph);
```

Comment:
It used to pass parent-children relationship and the root node. Now `DependencyGraph` knows both.